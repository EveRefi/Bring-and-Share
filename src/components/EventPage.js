// list of things to import
import React, { useState, useEffect } from "react";

import AddParticipant from "./AddParticipant";
import "../styles/EventPage.css";
import axios from "axios";
import Participants from "./Participants";
import HostDetails from "./HostDetails";
import NavBar from "./NavBar";

const EventPage = (props) => {
  
  const eventId = props.match.params.eventId;

  const [eventName, setEventName] = useState("");
  const [hostName, setHostName] = useState("");
  const [eventDate, setEventDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventAddress, setEventAddress] = useState("");
  const [hostEmail, setHostEmail] = useState("");
  const [fetchParticipants, setFetchParticipants] = useState(true);
  const [participants, setParticipants] = useState([]);
  const eventLink = `https://bring-and-share-silk.vercel.app/events/${eventId}`;
  useEffect(() => {
    async function fetchData() {
      await axios
        .get(`https://final-mcrcodes-project.herokuapp.com/events/${eventId}`)
        .then((response) => {
          setEventName(response.data.eventName);
          setHostName(response.data.hostName);
          setHostEmail(response.data.hostEmail);
          setEventDate(response.data.date);
          setEventTime(response.data.time);
          setEventAddress(response.data.address);
       
        })
        .catch((error) => {
          console.log(error);
        });
      
    }
  
    fetchData();
  }, [eventId]);

  return (
    <>
      <div data-testid="eventdetails">
        <h2>Here is your amazing page</h2>
        <p>Have fun at your event!</p>
        <p>Share event link with your attendees:{eventLink}</p>

        <div className="eventDetails">
          <h3>Event details</h3>
          <p>Event ID: {eventId}</p>
          <p>Event Name: {eventName}</p>
          <p>Host Name: {hostName}</p>
          <p>Event Date: {eventDate}</p>
          <p>Event Time: {eventTime}</p>
          <p>Event Address: {eventAddress}</p>
          <p>Host Email: {hostEmail}</p>
        </div>
        <HostDetails eventId={eventId} />
        <Participants
          eventId={eventId}
          fetchParticipants={fetchParticipants}
          setFetchParticipants={setFetchParticipants}
          
        />
        <AddParticipant
          eventId={eventId}
          fetchParticipants={fetchParticipants}
          setFetchParticipants={setFetchParticipants}
        />
        <NavBar 
         eventId={eventId}
        />
      </div>
    </>
  );
};

export default EventPage;
