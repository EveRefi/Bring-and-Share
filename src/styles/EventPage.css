.eventDetails {
  border: 5px solid yellow;
}
