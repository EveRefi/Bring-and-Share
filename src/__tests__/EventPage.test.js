import React from "react";
import { render } from "@testing-library/react";
import EventPage from "../components/EventPage";

describe('EventPage component', () => {
    const mockId = 1234

    it('it renders correctly', () => {

        const {asFragment} = render (<EventPage eventId={mockId}/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders Participants, and AddParticipant components', () => {
        const {getByTestId} = render (<EventPage eventId={mockId}/>)

        expect(getByTestId('eventdetails')).toBeInTheDocument()
    })


})