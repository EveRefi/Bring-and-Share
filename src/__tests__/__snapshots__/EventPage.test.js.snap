// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPage component it renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Here is your amazing page
    </h2>
    <p>
      Have fun at your event!
    </p>
    <div
      class="eventDetails"
    >
      <h3>
        Event details
      </h3>
      <p>
        Event ID: 1234
      </p>
      <p>
        Event Name: 
      </p>
      <p>
        Host Name: 
      </p>
      <p>
        Event Date: 
      </p>
      <p>
        Event Time: 
      </p>
      <p>
        Event Address: 
      </p>
    </div>
    <div
      class="participants"
    />
    <div
      class="addParticipant"
    >
      <h2>
        Add your participants
      </h2>
      <p>
        Fill in all fields and click "create event"
      </p>
      <form>
        <div
          class="name"
        >
          <label
            for="name"
          >
            Name 
            <br />
            <input
              data-testid="name"
              id="name"
              name="name"
              placeholder="Name of fellow Party Queen..."
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="toBring"
        >
          <label
            for="toBring"
          >
            I am bringing... 
            <br />
            <input
              data-testid="toBring"
              id="toBring"
              name="toBring"
              placeholder="Ferrero Rocher"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="dislikes"
        >
          <label
            for="dislikes"
          >
            I don't like... 
            <br />
            <input
              data-testid="dislikes"
              id="dislikes"
              name="dislikes"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="dietInfo"
        >
          <label
            for="dietInfo"
          >
            Diet info 
            <br />
            <input
              data-testid="dietInfo"
              id="dietInfo"
              name="dietInfo"
              placeholder="Vegetarian? Kosher? Coeliac?"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="createEventButton"
        >
          <button
            class="addButton"
            data-testid="addButton"
            type="submit"
          >
            Add
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
